--- uisp-20050207/src/Avr.C.old	2005-02-07 23:55:08.000000000 +0100
+++ uisp-20050207/src/Avr.C	2005-12-22 17:55:56.000000000 +0100
@@ -65,7 +65,7 @@
   { "ATtiny22",     0x91, 0x06,   2048,   0,  128,  4000, 4000, AVR_TN22 },
 
   { "ATtiny26",     0x91, 0x09,   2048,  32,  128,  4500, 9000, AVR_TN26 },
-  { "ATtiny2313",   0x91, 0x0A,   2048,   0,  128,  4000, 4000, AVR_TN2313 },
+  { "ATtiny2313",   0x91, 0x0A,   2048,  32,  128,  4000, 4000, AVR_TN2313 },
 
 #if 0
   /* 12V parallel programming only; here just for the evidence */
